hyracks/hyracks-dataflow-std/src/main/java/org/apache/hyracks/dataflow/std/structures/SerializableVector.java

```diff
@@ -15,37 +41,71 @@ public SerializableVector(IResetableComparableFactory factory, int recordSize){
      * @param frameSize
      * @param recordSize
      */
-    public SerializableVector(IResetableComparableFactory factory, int frameSize, int recordSize){
-
+    public SerializableVector(int recordSize, int frameSize){
+        frames = new ArrayList<>();
+        init(recordSize, frameSize);
     }
 
     @Override
-    public boolean get(int index, IResetableComparable record) {
-        return false;
+    public boolean get(int index, IResetableSerializable record) {
+        if(index >= sVectorSize) {
+            throw new IndexOutOfBoundsException("index out of bound");
+        }
+        int frameIdx = getFrameIdx(index);
+        int offsetInFrame = getOffsetInFrame(index);
+        record.deserialize(frames.get(frameIdx), offsetInFrame, recordSize);
+        return true;
```

Comment:
if it always return true, we should change the interface from `boolean` to `void`